src/schemas/generator/v1/rules.ts

```diff
@@ -72,11 +72,25 @@ export const StatusCodeSelectorSchema = z.object({
   type: z.literal('status-code'),
 })
 
-export const SelectorSchema = z.discriminatedUnion('type', [
+export const TextSelectorSchema = z.object({
+  type: z.literal('text'),
+  from: z.enum(['headers', 'body', 'url']),
+  value: z.string(),
+})
+
+export const ExtractorSelectorSchema = z.discriminatedUnion('type', [
+  BeginEndSelectorSchema,
+  RegexSelectorSchema,
+  JsonSelectorSchema,
+  HeaderNameSelectorSchema,
+])
```

Comment:
Split `Selector` schema into `Extractor` and `Replacer`, because `Text` selector doesn't apply to extractions